### PR TITLE
Revert when loading 0 size typed storagevec

### PR DESCRIPTION
**Pull request by SwayStar123**
## Description
Trying to allocate 0 bytes results in a reverts. This pr reverts earlier intentionally and adds documentation about it

## Checklist

- [x] I have linked to any relevant issues.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation where relevant (API docs, the reference, and the Sway book).
   - [x] If my change requires substantial documentation changes, I have [requested support from the DevRel team](https://github.com/FuelLabs/devrel-requests/issues/new/choose)
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added (or requested a maintainer to add) the necessary `Breaking*` or `New Feature` labels where relevant.
- [ ] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [ ] I have requested a review from the relevant team or maintainers.

### CODE REVIEW

File: sway-lib-std/src/storage/storage_vec.sw [hunk 1]
```diff
@@ -851,10 +851,18 @@ impl<V> StorageKey<StorageVec<V>> {
 
     /// Load a `Vec` from the `StorageVec`.
     ///
+    /// # Additional Information
+    ///
+    /// This method does not work for any V type that has a 0 size, such as StorageVec itself. Meaning you cannot use this method on a StorageVec<StorageVec<T>>.
```
Comment (K1-R1) [inline sway-lib-std/src/storage/storage_vec.sw:856]:
```suggestion
    /// This method does not work for any `V` type that has a 0 size, such as StorageVec itself. Meaning you cannot use this method on a StorageVec<StorageVec<T>>.
```

**Review by K1-R1 — DISMISSED** [review]

**Review by IGI-111 — APPROVED** [review]

File: test/src/sdk-harness/test_projects/storage_vec_nested/mod.rs [hunk 2]
```diff
@@ -33,3 +33,11 @@ async fn nested_vec_access_insert() {
...
     methods.nested_vec_access_insert().call().await.unwrap();
 }
+
+#[tokio::test]
+#[should_panic]
+async fn revert_on_load() {
+    let methods = test_storage_vec_nested_instance().await.methods();
+
+    methods.revert_on_load().call().await.unwrap();
+}
```
Comment (bitzoic) [inline test/src/sdk-harness/test_projects/storage_vec_nested/mod.rs:43]: Please make sure you run the formatter before committing.

File: sway-lib-std/src/storage/storage_vec.sw [hunk 3]
```diff
@@ -851,10 +851,18 @@ impl<V> StorageKey<StorageVec<V>> {
 
     /// Load a `Vec` from the `StorageVec`.
     ///
+    /// # Additional Information
+    ///
+    /// This method does not work for any `V` type that has a 0 size, such as StorageVec itself. Meaning you cannot use this method on a StorageVec<StorageVec<T>>.
```
Comment (bitzoic) [inline sway-lib-std/src/storage/storage_vec.sw:856]:
```suggestion
    /// This method does not work for any `V` type that has a 0 size, such as `StorageVec` itself. Meaning you cannot use this method on a `StorageVec<StorageVec<T>>`.
```

File: test/src/sdk-harness/test_projects/storage_vec_nested/src/main.sw [hunk 4]
```diff
@@ -47,6 +50,15 @@ impl ExperimentalStorageTest for Contract {
 
         // test_access(inner_vec0, inner_vec1, inner_vec2);
     }
+
+    #[storage(read, write)]
+    fn revert_on_load() {
```
Comment (bitzoic) [inline test/src/sdk-harness/test_projects/storage_vec_nested/src/main.sw:55]:
```suggestion
    fn revert_on_load_self() {
```
```suggestion
    fn revert_on_load_storage_vec() {
```
```suggestion
    fn revert_on_load_type_size_zero() {
```
This name doesn't properly reflect what it is testing

**Review by IGI-111 — APPROVED** [review]

**Review by K1-R1 — APPROVED** [review]

**Pull request merged**